src/Engine/Scene/Scene.hpp

```diff
@@ -79,24 +77,18 @@ class Scene {
 
 };
 
-
-
 // TEMPLATE IMPLEMENTATION /////////////////////////////////////////////////////
 
-
-
-template <typename CompT, typename... Args>
-CompT & Scene::createComponent(Args &&... args) {
-    return addComponent(std::unique_ptr<CompT>(new CompT(std::forward<Args>(args)...)));
-}
-
 template <typename CompT>
 CompT & Scene::addComponent(std::unique_ptr<CompT> component) {
     CompT & comp(*component);
-    m_componentInitQueue[std::type_index(typeid(CompT::SystemClass))].emplace_back(std::move(component));
+    m_componentInitQueue[std::type_index(typeid(typename CompT::SystemClass))].emplace_back(std::move(component));
```

Comment:
why did you add typename here?

Reply:
Don't know, this is the error I get when it is not there

error: dependent-name ‘CompT:: SystemClass’ is parsed as a non-type, but instantiation yields a type

Reply:
huh, what compiler are you using?

Reply:
i made that change across the board, hopefully shouldn't have any other issues

Reply:
thanks. g++